EIPS/eip-6051.md

```diff
@@ -0,0 +1,469 @@
+---
+eip: 6051
+title: Private Key Encapsulation
+description: defines a specification for encapsulating private keys.
+author: Base Labs (@Base-Labs), Weiji Guo (@weiji-cryptonatty)
+discussions-to: https://ethereum-magicians.org/t/private-key-encapsulation-to-move-around-securely-without-entering-seed/11604
+status: Draft
+type: Standards Track
+category: ERC
+created: 2022-11-21
+---
+
+
+## Abstract
+
+This EIP proposes a mechanism to encapsulate a private key so that it could be securely relocated to another application without providing the seed. This EIP combines `ECIES` (Elliptic Curve Integrated Encryption Scheme) and optional signature verification under various choices to ensure that the private key is encapsulated to a known or trusted party.
+
+## Motivation
+
+There are various cases that we might want to export one of many private keys from a much securer but less convenient wallet, which is controlled with a seed or pass phrase.
+
+1.  We might dedicate one of many private keys for messaging purpose, and that private key is probably managed in a not-so-secure manner;
+2.  We might want to export one of many private keys from a hardware wallet, and split it with MPC technology so that a 3rd party service could help us identify potential frauds or known bad addresses, enforce 2FA, etc., meanwhile we can initiate transactions from a mobile device with much better UX and without carrying a hardware wallet.
```

Comment:
Reviewing the motivation, however, it seems in this exact case we may have alternative request to R2 and R3, that is, each party $P_i$ in a MPC setting may generate its own ephemeral key pair $(r_i, R_i)$, and then Sender application may safely split `sk` safely into several pieces $sk_i$, then encapsulate each $sk_i$ to $R_i$. This effectively replaces the usual Distributed Key Generation procedure in a typical MPC protocol. 

To be fair, this may be added as an amendment to EIP-6051. I need to survey MPC protocols to see if the above suggestions work in their security models.